File: server/recorder/src/main/java/org/rliz/cfm/artist/model/Artist.java
package org.rliz.cfm.artist.model;

import org.rliz.cfm.common.model.AbstractEntity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.validation.constraints.NotNull;
import java.util.UUID;

/**
 * Represents an artist.
 */
@Entity
public class Artist extends AbstractEntity {

    public UUID mbid;

    @Column
    @NotNull
    public String name;

    /**
     * Default constructor.
     */
    public Artist() {
        // for JPA
    }

    /**
     * Field setting constructor.
     * @param mbid the musicbrainz ID of this artist, can be null
     * @param name the name of this artist
     */
    public Artist(UUID mbid, String name) {
        this.mbid = mbid;
        this.name = name;
    }
}

File: server/recorder/src/main/kotlin/org/rliz/cfm/recorder/artist/data/Artist.kt
package org.rliz.cfm.recorder.artist.data

import org.rliz.cfm.recorder.common.data.AbstractModel
import java.util.*
import javax.persistence.Entity
import javax.persistence.Index
import javax.persistence.Table
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
@Table(
        indexes = arrayOf(
                Index(
                        name = "IX_Artist_Identifier",
                        columnList = "uuid",
                        unique = true
                ))
)
class Artist : AbstractModel {

    @NotNull
    @Size(min = 1)
    var name: String? = null

    var lastUpdated: Long? = null

    constructor() : super()

    constructor(uuid: UUID, name: String, lastUpdated: Long?) : super(uuid) {
        this.name = name
        this.lastUpdated = lastUpdated
    }
}
